core/src/input_reader/guest_input_reader.rs

```diff
@@ -485,10 +551,21 @@ fn get_balance_churn_limit(spec: &ChainSpec, total_active_balance: u64) -> Resul
     churn.safe_sub(churn.safe_rem(spec.effective_balance_increment)?)
 }
 
-/// Extracts an u64 from a 32-byte SSZ chunk.
-fn u64_from_chunk(node: &[u8; 32]) -> u64 {
-    assert!(node[8..].iter().all(|&b| b == 0));
-    u64::from_le_bytes(node[..8].try_into().unwrap())
+trait U64ChunkPos {
+    fn as_range() -> std::ops::Range<usize>;
+}
```

Comment:
I am always a bit confused by that. I thought that SSZ merkelization packs several smaller types into one chunk, but obviously this is not the case, we get one chunk per field with the first bytes set...